### PR TITLE
Refactor weight constants, Update to remove redundant overhead

### PR DESCRIPTION
**Pull request by shawntabrizi**
This PR moves some of the FRAME specific weight constants (DbWeight, ExtrinsicBaseWeight, BlockExecutionWeight) to the `frame-support` library. Since we already make heavy assumptions about how to derive these weights based on our specific setup of FRAME, it makes sense to move these thing into there.

Additionally, this updates some weight values across various extrinsic to make them even a bit more lightweight!

We found that we had double counted the overhead implied by a Heavy DB by testing our DB Trie stats with a full DB and also testing our runtime with a full chain spec. The full chain spec in the runtime was not needed since we assume now every DB read will be ~200K item (5 trie look ups) heavy.

As a result, we get some lower weights!

This updates:
* Balances
* Vesting
* Utility

TODO in another PR: Update all weights to use these consts

### CODE REVIEW

File: frame/support/src/weights.rs [hunk 1]
```diff
@@ -141,6 +141,32 @@ pub use sp_runtime::transaction_validity::TransactionPriority;
 /// Numeric range of a transaction weight.
 pub type Weight = u64;
 
+/// These constants are specific to FRAME, and the current implementation of its various components.
+/// For example: FRAME System, FRAME Executive, our FRAME support libraries, etc...
+pub mod constants {
+	use super::*;
```
Comment (kianenigma) [inline frame/support/src/weights.rs:147]:
```suggestion
	use super::{Weight, RuntimeDbWeight};
```

File: frame/support/src/weights.rs [hunk 2]
```diff
@@ -141,6 +141,32 @@ pub use sp_runtime::transaction_validity::TransactionPriority;
...
+	use crate::parameter_types;
+
```
Comment (kianenigma) [inline frame/support/src/weights.rs:149]:
```suggestion
	pub const WEIGHTS_PER_MILLI_SECOND = 1_000_000_000;
	pub const WEIGHTS_PER_SECOND = 1000 * WEIGHTS_PER_MILLI_SECOND; 
```

File: bin/node/runtime/src/lib.rs [hunk 3]
```diff
@@ -120,17 +120,9 @@ parameter_types! {
 	pub const BlockHashCount: BlockNumber = 250;
 	/// We allow for 2 seconds of compute with a 6 second average block time.
 	pub const MaximumBlockWeight: Weight = 2_000_000_000_000;
```
Comment (kianenigma) [inline bin/node/runtime/src/lib.rs:122]:
```suggestion
	pub const MaximumBlockWeight: Weight = 2 * WEIGHTS_PER_SECOND;
```

Comment (kianenigma) [reply bin/node/runtime/src/lib.rs:122]: If you do this, then even if we decide to scale all the weights again (to for example 1 weight = 1 nano-seconds) then things will not break here again. 

If you want to make it even fancier, you could impl `ops::Add` and `ops::Mul` for `Get<_>` and then we can define the `MaximumBlockWeight ` inside `parameter_type` in `weights::constants` and then have it very nicely here or in any other runtime:
impl frame_system::Trait for Runtime {
     type MaximumBlockWeight = 2 * WeightUnitsPerSecond;
}
```

but a simple const is also already good enough.

**Comment by kianenigma** [comment]
probably will need a companion as well.

**Comment by gavofyork** [comment]
apparently not.

**Comment by shawntabrizi** [comment]
@gavofyork will make a similar change to polkadot, but ya, this merging wont break anything

**Comment by JoshOrndorff** [comment]
@shawntabrizi I know this is old, but I have a question.

Moving the `ExtrinsicBaseWeight` to frame support implies (to me at least) that it should be re-useable across chains. But doesn't it depend on what signature scheme your chain uses? When you benchmarked it, did you use only sr25519, or also ed25519 and ecdsa?

Do you expect chains to re-benchmark this themselves?

**Comment by shawntabrizi** [comment]
@JoshOrndorff yes, we provide the "baseline" implementation, but if you make changes to the assumptions, you will need to redo the benchmarking.

Changing the signature scheme could def cause a change here.

The docs for how we generated that number can be found here: https://www.shawntabrizi.com/substrate-graph-benchmarks/docs/#/extrinsic

**Pull request merged**